python/core/auto_generated/processing/qgsprocessing.sip.in

```diff
@@ -52,6 +52,11 @@ Converts a source ``type`` to a string representation.
 %End
 
     static const QString TEMPORARY_OUTPUT;
```

Comment:
Why are these part of public api?

Reply:
Sorry, this comment is supposed to be attached to the new API added to the news feed parser (but applies to all the settings which use qsettings to store internal details)

Reply:
just to be sure we talk about the same thing, the settings will indeed be part of the API, so one can access them.

Reply:
Ok, I have a fundamental issue with this concept (settings as API). I have two issues here:

1. Many settings (such as the ones in the news feed parser) are never intended to be changed by users. They are using qsettings as a storage mechanism for persistent values only. This also extends to things like use of qsettings for storing "last used folder", widget geometries, etc. None of that should EVER be exposed as public api.
2. Some settings are arguably more "API"-ish, But even still I question whether we WANT to advertise these as stable API. Doing so really constricts us from reworking the corresponding areas of code in future, as this often requires changing settings configuration or removing existing settings. So how stable API are we making settings? Can I remove one in future and cause plugins to raise a python exception? Do we leave them forever but just start ignoring them when we don't want that setting anymore? We need much more clarity here before wholesale adding all settings as public stable API.
3. From a user perspective, it's **extremely** annoying when plugins change user settings eg "default crs for new layers", and potentially results in a broken qgis user experience (and yes, there are plugins in the repo which change the default user crs to a hardcoded specific projection on startup!!). 

Now I realise that plugins can already use raw qsettings to modify settings. But we've never advertised this as supported in any way, so doing so has always been a "use at own risk" thing for plugin authors.

Reply:
I think your concerns are all valid. I propose following changes:
1. Exclude all settings definitions form the python API by putting them in `#ifndef SIP_RUN`.
2. Settings remain in public section of the class where they are defined. Two reasons for that:
   -  They are often used from more than one class
   -  Otherwise would not be possible to add them to the core register (as it is done now).
3. Exclude QgsSettingsRegistry core from the python API.
4. New interface in QgsApplication to add/remove QgsSettingsRegistry. This would give to plugins the possibility to register their own settings registry and have settings available in the advanced settings editor.
 
What do you think?

Reply:
@domi4484 

> Exclude QgsSettingsRegistry core from the python API.

I wouldn't go this far. Rather I'd say initially we don't expose any settings definitions (as you've mentioned in (1) ), but leave the option to selectively expose settings which we think SHOULD be exposed. I.e. flip it from being opt-out to opt-in.

Reply:
Ok I have just put all settings in `#ifndef SIP_RUN`in last commit.